driver/src/eth_rpc.rs

```diff
@@ -0,0 +1,32 @@
+use crate::contracts;
+
+use ethcontract::web3::error::Error;
+use ethcontract::web3::futures::Future;
+use ethcontract::web3::types::TransactionReceipt;
+use ethcontract::H256;
+#[cfg(test)]
+use mockall::automock;
+
+/// Interface defining possible interactions via Ethereum RPCs
+/// Main purpose it to allow mocking the concrete Web3 implementation
+#[cfg_attr(test, automock)]
+pub trait EthRpc {
```

Comment:
Would it make sense for `ethcontract` to return the tx receipt? It already queried it to read the failure.

Reply:
Yes that would make sense. This way we'd save one round trip.